server/src/main/java/org/elasticsearch/common/cli/EnvironmentAwareCommand.java

```diff
@@ -153,4 +155,9 @@ private static void putSystemPropertyIfSettingIsMissing(
     /** Execute the command with the initialized {@link Environment}. */
     public abstract void execute(Terminal terminal, OptionSet options, Environment env, ProcessInfo processInfo) throws Exception;
 
+    protected SecureSettingsLoader secureSettingsLoader(Environment env) {
```

Comment:
EnvironmentAwareCommand seems really broad. Can we narrow this to a utility method shared by ServerCli and WindowsServiceDaemon? Perhaps in SecureSettingsLoader itself?

Reply:
Ideally it should be in SecureSettingsLoader, but it would need to be static, which is harder to mock. Let me think if I can make tests work still.